src/components/application_manager/src/commands/mobile/system_request.cc

```diff
@@ -43,9 +45,356 @@ Copyright (c) 2013, Ford Motor Company
+                  << app_data[os_type][json::urlScheme].asString().length()
+                  << "]>[" << max_url_scheme_lengh_ << "]");
+          return false;
+        }
+      }
+      if (os_type.empty()) {
+        if (app_data.keyExists(json::android)) {
+          os_type = json::android;
+          if (!app_data[os_type].keyExists(json::packageName)) {
+            LOG4CXX_WARN(logger_,
+                         error_prefix_
+                             << "Can't find package name in json file.");
+            return false;
+          }
+          if (app_data[json::android][json::packageName].asString().length() >
+              max_package_name_length_) {
+            LOG4CXX_WARN(logger_,
+                         error_prefix_
+                             << "Package name length ["
+                             << app_data[json::android][json::packageName]
+                                    .asString()
+                                    .length() << "] exceeds max length ["
+                             << max_package_name_length_ << "]in json file.");
+            return false;
+          }
+        }
+      }
+
+      if (os_type.empty()) {
+        LOG4CXX_WARN(logger_,
+                     error_prefix_
+                         << "Can't find mobile OS type in json file.");
+        return false;
+      }
+
+      // Languages verification
+      if (!app_data[os_type].keyExists(json::languages)) {
+        LOG4CXX_WARN(logger_, error_prefix_ << "\"languages\" not exists");
+        return false;
+      }
+      if (!ValidateLanguages(app_data[os_type][json::languages],
+                             synonyms_map)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  bool ValidateAppIdAndAppName(const smart_objects::SmartObject& app_data,
+                               std::set<std::string>& app_ids_set) const {
+    // Verify appid
+    if (!app_data.keyExists(json::appId)) {
+      LOG4CXX_WARN(logger_, error_prefix_ << "Can't find app ID in json file.");
+      return false;
+    }
+    bool result = true;
+    // Verify appid lenght
+    const std::string app_id(app_data[json::appId].asString());
+    if (app_id.length() > max_app_id_length_) {
+      LOG4CXX_WARN(logger_,
+                   error_prefix_
+                       << "An Object ID length exceeds maximum allowed ["
+                       << app_id.length() << "]>[" << max_app_id_length_
+                       << "]");
+      result = false;
+    }
+
+    // Verify that appid is unique
+    if (app_ids_set.find(app_id) != app_ids_set.end()) {
+      LOG4CXX_WARN(logger_,
+                   error_prefix_ << "An Object ID is not unigue [" << app_id
+                                 << "]");
+      result = false;
+    } else {
+      app_ids_set.insert(app_id);
+    }
+
+    // Verify that app is not registered yet
+    ApplicationSharedPtr registered_app =
+        ApplicationManagerImpl::instance()->application_by_policy_id(app_id);
```

Comment:
@okoreniak  Please try to avoid singleton usage, You can Pass ApplicationManagerImpl as parameter to constructor of this class